src/router_core/modules/edge_router/addr_proxy.c

```diff
@@ -254,7 +254,7 @@ static void on_conn_event(void *context, qdrc_event_t event, qdr_connection_t *c
                 // If the address has more than zero attached destinations, create an
                 // incoming link from the interior to signal the presence of local consumers.
                 //
-                if (DEQ_SIZE(addr->rlinks) > 0) {
+                if (DEQ_SIZE(addr->rlinks) > 0 || (DEQ_SIZE(addr->subscriptions) > 0 && addr->propagate_local)) {
                     if (DEQ_SIZE(addr->rlinks) == 1) {
                         //
                         // If there's only one link and it's on the edge connection, ignore the address.
```

Comment:
Q: should an inlink be added unconditionally if there's a propagate subscription? IOW, does the addr->rlinks == 1... precondition apply in that case?